src/asciidoc/core-beans.adoc

```diff
@@ -4562,14 +4562,16 @@ automatically resolved, with no special setup necessary.
 	}
 ----
 
-[NOTE]
+[WARNING]
 ====
-`@Autowired`, `@Inject`, `@Resource`, and `@Value` annotations are handled by Spring
+`@Autowired`, `@Inject`, `@Resource`, and `@Value`  annotations are handled by Spring
 `BeanPostProcessor` implementations which in turn means that you __cannot__ apply these
 annotations within your own `BeanPostProcessor` or `BeanFactoryPostProcessor` types (if
 any). These types must be 'wired up' explicitly via XML or using a Spring `@Bean` method.
-====
 
+Also, do not use `@Autowired`, `@Inject`, and `@Resource` annotations within a BeanFactory class nor any of its dependencies. Doing so can result in a `BeanCurrentlyInCreationException`. The `@Value` annotation may also cause the problem, d
+epending on whether it causes a bean to be instantiated.
+====
```

Comment:
Good find digging up those tickets. 

One of tickets SPR-8704 shows that it was turned into a documentation task and documentation updates were applied on this topic already including both master https://github.com/spring-projects/spring-framework/commit/fd41f63ec065b890a539451532392fa9b5bb0b28 and 4.3 https://github.com/spring-projects/spring-framework/commit/28e114b569d0e3e19ea0e0160704b6726810f428.

I think the explanation here is not very good because explaining things in terms of the exception doesn't help much and made me go lookup the javadoc for that exception which didn't bring a more satisfying answer either:
> Doing so can result in a `BeanCurrentlyInCreationException`

I find this explanation under the ticket much more useful because it tells me why I shouldn't do it (interface vs annotation based dependency injection):
>  we tend to generally recommend against using annotation-driven injection in FactoryBean implementations. Since such classes implement an interface-based contract to begin with, I would rather design them to use programmatic BeanFactory lookups for the target beans needed.

@jhoeller from the above linked tickets it seems that quite a few people are running into this and documentation alone is not effective in preventing time wasted debugging. If there is anything that could be done to detect or proactively reject..